src/pkg/bundler/remote.go

```diff
@@ -31,10 +34,11 @@ type RemoteBundler struct {
 	RemoteSrc       *oci.OrasRemote
 	RemoteDst       *oci.OrasRemote
 	localDst        *ocistore.Store
+	tmpDir          string
 }
 
 // NewRemoteBundler creates a bundler to pull remote Zarf pkgs
-func NewRemoteBundler(pkg types.BundleZarfPackage, url string, localDst *ocistore.Store, remoteDst *oci.OrasRemote) (RemoteBundler, error) {
+func NewRemoteBundler(pkg types.BundleZarfPackage, url string, localDst *ocistore.Store, remoteDst *oci.OrasRemote, tmpDir string) (RemoteBundler, error) {
```

Comment:
The logic here is hidden and not documented, and it is used three different ways. For code cleanup / simplicity, I recommend breaking this into three methods or providing some usage notes

Reply:
100% ! I think the `bundler` pkg itself is probably wrong. There's a matrix of operations to consider when doing local + remote bundles that interact with local + remote packages. TLDR: yes, def needs a refactor. In the interim I'll make a note to add comments indicating what happens when you pass the various args